Lib/posixpath.py

```diff
@@ -396,7 +374,8 @@ def normpath(path):
 def abspath(path):
     """Return an absolute path."""
     path = os.fspath(path)
-    if not isabs(path):
+    sep = _get_sep(path)
```

Comment:
`_get_sep()` does an `isinstance(path, bytes)` check, which is also performed below. I think we can reduce that to one call:

```python
def abspath(path):
    """Return an absolute path."""
    path = os.fspath(path)
    if isinstance(path, bytes):
        if path[:1] != b'/':
            path = join(os.getcwdb(), path)
    else:
        if path[:1] != '/':
            path = join(os.getcwd(), path)
    return normpath(path)
```

Reply:
You should also be able to replace `join(os.getcwd(), path)` with `f'{os.getcwd()}/{path}'`, which might save a chunk of time.